### PR TITLE
Miscellaneous JS Fixes

### PR DESCRIPTION
**Pull request by alecpm**
Some JS cleanup. Add pre-commit hooks to validate YAML and run JS build.

### CODE REVIEW

File: dlgr/griduniverse/static/scripts/demo.js [hunk 1]
```diff
@@ -1078,7 +1076,7 @@ function onGameStateChange(msg) {
   updateDonationStatus(state.donation_active);
 
   // Update gridItems
-  if (state.items !== undefined && state.items !== null) {
+  if (! _.isNil(state.items)) {
```
Comment (jessesnyder) [inline dlgr/griduniverse/static/scripts/demo.js:1079]: I think this is `isNull()` ? https://underscorejs.org/#isNull

File: dlgr/griduniverse/static/scripts/dist/.gitattributes [hunk 2]
```diff
@@ -0,0 +1,4 @@
+*.js -diff -merge
```
Comment (jessesnyder) [inline dlgr/griduniverse/static/scripts/dist/.gitattributes:1]: I must have just had the file in the wrong place(?) I had it at the repository root.

**Review by jessesnyder — APPROVED** [review]
Ahhhh... Looks great.

I think `_.isNil` should be `_.isNull` unless there's something I didn't see in the docs.

File: dlgr/griduniverse/static/scripts/demo.js (hunk 1 above)
Comment (alecpm) [reply dlgr/griduniverse/static/scripts/demo.js:1079]: We're using lodash, so it's [isNil](https://lodash.com/docs/4.17.15)

File: dlgr/griduniverse/static/scripts/dist/.gitattributes (hunk 2 above)
Comment (alecpm) [reply dlgr/griduniverse/static/scripts/dist/.gitattributes:1]: It's OK to have in the repository root, but then the glob paths are more difficult I think,

**Pull request merged**